subsystem/drivetrain/swerve_module.py

```diff
@@ -0,0 +1,339 @@
+            "steer_motor": channel_base + 1,
+            "absolute_encoder": channel_base + 2
+        }
+
+        # Physical device instantiation
+        self.drive_motor = rev.SparkMax(self.id_lookup["drive_motor"] , rev.SparkLowLevel.MotorType.kBrushless)
+        self.steer_motor = rev.SparkMax(self.id_lookup["steer_motor"], rev.SparkLowLevel.MotorType.kBrushless)
+        self.absolute_encoder = phoenix6.hardware.CANcoder(self.id_lookup["absolute_encoder"])
+
+        self.drive_motor_encoder = self.drive_motor.getEncoder()
+        self.steer_motor_encoder = self.steer_motor.getEncoder()
+
+        self.drive_motor_pid = self.drive_motor.getClosedLoopController()
+        self.steer_motor_pid = self.steer_motor.getClosedLoopController()
+
+        # Configuration setup
+        self.drive_motor_config = rev.SparkMaxConfig()
+        self.steer_motor_config = rev.SparkMaxConfig()
+
+        # Absolute encoder configuration
+        absolute_encoder_configurator = self.absolute_encoder.configurator
+        absolute_encoder_config = phoenix6.configs.CANcoderConfiguration()
+        (
+            absolute_encoder_config.magnet_sensor
+                .with_absolute_sensor_discontinuity_point(1)
+                .with_magnet_offset(encoder_calibration)
+                .with_sensor_direction(phoenix6.signals.SensorDirectionValue.COUNTER_CLOCKWISE_POSITIVE)
+        )
+
+        status = absolute_encoder_configurator.apply(absolute_encoder_config, 0.25)
+        if not status.is_ok():
+            raise RuntimeError(
+                f"Failed to configure CAN encoder on id {self.id_lookup['absolute_encoder']}. Error {status}"
+            )
+
+        status = self.absolute_encoder.get_position().set_update_frequency(self.constants.kCanStatusFrameHz, 0.25)
+        if not status.is_ok():
+            raise RuntimeError(
+                f"Failed to configure CAN encoder on id {self.id_lookup['absolute_encoder']}. Error {status}"
+            )
+
+        # Steer motor configuration
+        self.instantiate_steer_config(invert_steer)
+        self.apply_motor_config(to_drive=False, burn_flash=True)
+
+        # Drive motor configuration
+        self.instantiate_drive_config(invert_drive)
+        self.apply_motor_config(to_drive=True, burn_flash=True)
+
+        # Baseline relative encoders
+        self.baseline_relative_encoders()
+
+    def instantiate_steer_config(self, invert: bool) -> None:
+        """
+        Set a starting configuration for the SparkMax controlling the steer motor. This configuration
+        is applied once the robot is turned on and will be held through power cycles unless otherwise updated.
+
+        Args:
+            invert: if True, flip the rotation direction that corresponds to the polarity of the steer motor input
+
+        Returns:
+            None - internal steer configuration is updated in-place
+        """
+        configureSparkMaxCanRates(self.steer_motor_config, drive_motor_flag=False)
```

Comment:
```suggestion
        sparkMaxUtils.configureSparkMaxCanRates(self.steer_motor_config, drive_motor_flag=False)
```
Should be able to update to use this now from raptacon3200

Reply:
Done by Chris